### PR TITLE
Jason

### PR DESCRIPTION
**Pull request by chirantan31**

### CODE REVIEW

File: TaskEngine/Tasks/DownloadMediaTask.cs [hunk 1]
```diff
@@ -34,14 +39,16 @@ protected override async Task OnConsume(Media media)
...
+                case SourceType.Box: video = await DownloadBoxVideo(media); break;
             }
             using (var _context = CTDbContext.CreateDbContext())
             {
                 await _context.Videos.AddAsync(video);
                 await _context.SaveChangesAsync();
                 Console.WriteLine("Downloaded:" + video);
-                _convertVideoToWavTask.Publish(video);
-            }                
+                // _convertVideoToWavTask.Publish(video);
```
Comment (chirantan31) [inline TaskEngine/Tasks/DownloadMediaTask.cs]: uncomment this

File: TaskEngine/Tasks/DownloadMediaTask.cs [hunk 2]
```diff
@@ -34,14 +39,16 @@ protected override async Task OnConsume(Media media)
                 case SourceType.Echo360: video = await DownloadEchoVideo(media); break;
                 case SourceType.Youtube: video = await DownloadYoutubeVideo(media); break;
                 case SourceType.Local: video = await DownloadLocalPlaylist(media); break;
+                // jason
```
Comment (chirantan31) [inline TaskEngine/Tasks/DownloadMediaTask.cs]: remove this

File: TaskEngine/Tasks/DownloadMediaTask.cs [hunk 3]
```diff
@@ -108,5 +115,75 @@ public async Task<Video> DownloadLocalPlaylist(Media media)
...
+                request.AddHeader("Connection", "keep-alive");
+                request.AddHeader("Referer", $"https://uofi.app.box.com/2.0/files/{media.Id}/content");
+                request.AddHeader("Cookie", "box_visitor_id=5da4f447d00911.72030283");
+                request.AddHeader("Accept-Encoding", "gzip, deflate");
+                request.AddHeader("Postman-Token", "a02a657a-e03b-41b3-a747-6190a61f3bea,b759367e-377f-46a6-b96f-f3a78a984ad3");
+                request.AddHeader("Cache-Control", "no-cache");
+                request.AddHeader("Accept", "*/*");
+                request.AddHeader("User-Agent", "PostmanRuntime/7.18.0");
+                // TODO: figure out a way to refresh token
+                request.AddHeader("Authorization", $"Bearer {access_token}");
```
Comment (chirantan31) [inline TaskEngine/Tasks/DownloadMediaTask.cs]: Remove all the header that aren't needed, for example the ones to do with Postman

File: TaskEngine/Tasks/DownloadPlaylistInfoTask.cs [hunk 4]
```diff
@@ -118,5 +124,85 @@ public async Task<List<Media>> GetLocalPlaylist(Playlist playlist, CTDbContext _
...
+                    foreach (JObject jObject in jArray.Children())
+                    {
+                        newMedia.Add(new Media
+                        {
+                            JsonMetadata = jObject,
+                            SourceType = playlist.SourceType,
+                            PlaylistId = playlist.Id,
+                            UniqueMediaIdentifier = jObject["sha1"].ToString(),
+                            // Id used to download specific file
+                            Id = jObject["id"].ToString()
```
Comment (chirantan31) [inline TaskEngine/Tasks/DownloadPlaylistInfoTask.cs]: Delete this line

File: TaskEngineDebug/bin/netcoreapp3.0/refresh.json [hunk 5]
```diff
@@ -0,0 +1 @@
+{"access_token":"4InJV53lEKflshRTt9rUAM003EDkZ4DI","expires_in":3836,"restricted_to":[],"refresh_token":"uPJGbJnf0kg1JWyUoN8pNrYbve7zoQCPa6UzfzR5wGxmM7E8h3SyyHTk1m9D9qyh","token_type":"bearer"}
```
Comment (chirantan31) [inline TaskEngineDebug/bin/netcoreapp3.0/refresh.json]: Are these secrets?

File: TaskEngine/Tasks/DownloadPlaylistInfoTask.cs [hunk 6]
```diff
@@ -118,5 +124,85 @@ public async Task<List<Media>> GetLocalPlaylist(Playlist playlist, CTDbContext _
...
+        public async Task<List<Media>> GetBoxPlaylist(Playlist playlist, CTDbContext _context)
+        {
+            var path = System.IO.Path.Combine(Directory.GetCurrentDirectory(), "refresh.json");
+            JObject refresh_json = JObject.Parse(File.ReadAllText(@path));
+            String access_token = (String)refresh_json.SelectToken("access_token");
+
+            while (true)
+            {
+                // send request using RestSharp
+                var client = new RestClient("https://uofi.app.box.com/2.0/folders/88965021711");
```
Comment (chirantan31) [inline TaskEngine/Tasks/DownloadPlaylistInfoTask.cs]: Where are you passing in the box folder id?

**Comment by angrave** [comment]
Old PR no longer required. Not merging

**Pull request closed without merging**